repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeServlet.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.powerfilter
+
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.inject.{Inject, Named}
+import javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper, ResponseMode}
+
+@Named
+class ReposeServlet @Inject()(router: PowerFilterRouter) extends HttpServlet with StrictLogging {
```

Comment:
Just noting that the `PowerFilterRouter` (or whatever it becomes) will need to be made a Spring-managed bean for this to work.

Reply:
Apparently the current bean is a power filter router factory, i have no idea why.

Reply:
Ah. Somehow I did not notice that. Well that's crappy. Well, the router will change with the system model (since it uses the destinations therein). So ideally, I think the router would still be a bean, but it would have to be a bean that can be updated or replaced. We can probably drop the factory and the router can initialize itself as a bean with the services it needs, plus the cluster and node IDs so that it can determine destinations. Which means the system model comes into play somewhere. Not sure exactly how that looks, but... that should be a separate story anyway.